src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -117,6 +132,31 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
         blockExplorerUriTemplate,
       }
     },
+    onSetSafeSettings: (settings: SafeSettings) => {
+      const EIP_1271_SUPPORTED_SAFE_VERSION = '1.0.0'
+
+      if (typeof settings.offChainSigning === 'boolean' && gte(safe.version, EIP_1271_SUPPORTED_SAFE_VERSION)) {
```

Comment:
If we don't check whether it is a `boolean`, it is not possible to disable the off-chain signing.

Reply:
Setting the local state seems to work even when not checking for a type or does the RPC call return something other than boolean when an app wants to disable off-chain signing?

Reply:
[`SafeSettings` can be `Partial`](https://github.com/safe-global/safe-apps-sdk/blob/safe-settings/packages/safe-apps-sdk/src/types/rpc.ts#L101-L103). It's best we check for the presence, otherwise it could be set to `undefined`. Regarding disabling it, I think we should allow it as this RPC method was pre-emptively planned with the intention of future `SafeSettings`.

Reply:
I've altered this according to another review comment. We always set it now.